packages/access-client/src/agent-data.js

```diff
@@ -151,11 +151,10 @@ export class AgentData {
    */
   sessionProof() {
     for (const { delegation } of this.delegations.values()) {
-      // @ts-expect-error "key" does not exist in object, unless it's a session capability
       const cap = delegation.capabilities.find(
         (c) =>
           // @ts-expect-error "key" does not exist in object, unless it's a session capability
-          c.can === Access.session.can && c.nb?.key === this.principal.did()
+          c.can === Access.session.can && c?.nb?.key === this.principal.did()
```

Comment:
if `c` is undefined I think it will already blow up earlier here right (in the c.can === Access.session.can)

Reply:
I removed the `?` I added.